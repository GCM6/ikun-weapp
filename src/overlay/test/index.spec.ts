import path from 'node:path';
import simulate from 'miniprogram-simulate';

describe('overlay', () => {
  const KOverlay = simulate.load(path.resolve(__dirname, '../overlay'), 'k-overlay', {
    rootPath: path.resolve(__dirname, '../../'),
  });

  test('should render overlay add match snapshot', () => {
    const comp = simulate.render(KOverlay);
    comp.attach(document.createElement('parent-wrapper'));
    comp.setData({ mounted: true });

    expect(comp.toJSON()).toMatchSnapshot();
  });
});
