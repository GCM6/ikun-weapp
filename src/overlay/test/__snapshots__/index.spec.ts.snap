// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlay should render overlay add match snapshot 1`] = `
<wx-view
  ariaLabel="close"
  ariaRole="button"
  class="k-overlay  k-class"
  id=""
  style="z-index:11000;display: none"
  bind:tap="onTap"
  catch:touchmove="noop"
  bind:transitionend="onTransitionEnd"
/>
`;
